### PR TITLE
Overload import project method and add tests

### PR DESCRIPTION
**Pull request by rbarroso**
Overload import project method (project service) and add tests

### CODE REVIEW

File: activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java [hunk 1]
```diff
@@ -452,4 +447,31 @@ public void validateProject(Project project) {
 
         return validationErrors.stream();
     }
+
+    private Project importModelsFromProjectHolder(ProjectHolder projectHolder, String id){
```
Comment (mmartinadanx) [inline activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java]: I would get rid of this 'id' and log the error in the controller using this feature.

File: activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java [hunk 2]
```diff
@@ -293,27 +294,21 @@ private void convertZipElementToModelObject(ZipStream.ZipStreamEntry zipEntry, @
...
-        projectHolder.getModelJsonFiles().forEach(modelJsonFile -> {
-            importJSONModelFiles(projectHolder, createdProject, modelJsonFile);
-        });
-
-        projectHolder.getModelContentFiles().forEach(modelXmlFile ->
-                importXMLModelFiles(projectHolder, createdProject, modelXmlFile.getModelType(), modelXmlFile.getFileContent()));
-
-        Map<Model, FileContent> createdProcesses = createXMLModelFiles(projectHolder, createdProject);
-        createdProcesses.keySet().forEach(model -> updateModelProcessImported(projectHolder, model, createdProcesses.get(model)));
+        return importModelsFromProjectHolder(getProjectHolderFromZipStream(ZipStream.of(file), name), file.getOriginalFilename());
```
Comment (mmartinadanx) [inline activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java]: Does this ZipStream need to be closed?

**Review by mmartinadanx — CHANGES_REQUESTED** [review]
Well done! Just minor changes/comments

File: activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java (hunk 1 above)
Comment (rbarroso) [reply activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java]: Removed

File: activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java (hunk 2 above)
Comment (rbarroso) [reply activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java]: I don't think so. The zipstream is used below inside a try-with-resources statement

**Review by mmartinadanx — APPROVED** [review]

File: activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java [hunk 3]
```diff
@@ -293,27 +294,20 @@ private void convertZipElementToModelObject(ZipStream.ZipStreamEntry zipEntry, @
...
-
-        Map<Model, FileContent> createdProcesses = createXMLModelFiles(projectHolder, createdProject);
-        createdProcesses.keySet().forEach(model -> updateModelProcessImported(projectHolder, model, createdProcesses.get(model)));
+        return importModelsFromProjectHolder(getProjectHolderFromZipStream(ZipStream.of(file), name));
+    }
 
-        modelService.cleanModelIdList();
-        return createdProject;
+    /**
+     * Import an project form a zip inputstream.
```
Comment (pmartinezga) [inline activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java]:
```suggestion
     * Import a project from a zip inputstream.
```

**Review by pmartinezga — APPROVED** [review]
Just a typo

File: activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java (hunk 3 above)
Comment (rbarroso) [reply activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java]: Done. Thanks!

File: activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java (hunk 2 above)
Comment (erdemedeiros) [reply activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java]: From what I remember the good practice to follow is: "who opens the stream should take care of closing it". In this case, `ZipStream.of(file)` is opening the stream with `file.getInputStream()`, but it cannot close it, as the stream is used later on. So maybe it's better to completely remove `ZipStream.of(MultipartFile file)` and change it to:
```
try (InputStream inputStream = file.getInputStream()) {
            return importModelsFromProjectHolder(getProjectHolderFromZipStream(ZipStream.of(inputStream), name));
        }
```

**Review by erdemedeiros — APPROVED** [review]
LGTM!

**Pull request merged**